carbon-design-system/carbon-addons-iot-react
# 3016 list add pinned items

Closes #3016 

**Summary**

Added feature to lock rows and prevent them from being reordered

**Change List (commits, features, bugs, etc)**
- Added `lockedIds` prop to List and HierarchyList
- Added List story showing `lockedIds`
- Updated Hierarchy story to show knob for `lockedIds`

**Acceptance Test (how to verify the PR)**

1. Go to https://deploy-preview-3017--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-nested-reorder
2. Verify that Jeff McNeil can't be directly reordered and that there is a lock icon on the row
3. Turn on multi-nesting and make sure it looks and works ok
4. Turn on hasVirtualList and make sure it looks and works ok

1. Go to https://deploy-preview-3017--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-reorder-and-locked-rows
2. Verify that the two top rows can't be directly reordered and that there is a lock icon on the rows
3. Verify that it is not possible to drag and drop another row above the locked row
4. Turn on hasVirtualList and make sure it looks and works ok

**Regression Test (how to make sure this PR doesn't break old functionality)**

All normal List and Hierarchy List stories work as before



**Things to look for during review**
